extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/ImmutablePathMatcher.java

```diff
@@ -0,0 +1,346 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.BiConsumer;
+
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.quarkus.vertx.http.runtime.security.ImmutableSubstringMap.SubstringMatch;
+
+/**
+ * Handler that dispatches to a given handler based of a prefix match of the path.
```

Comment:
This comment is not really accurate given that it now supports more than prefix matches.

Reply:
Fixed.